### ISSUE TITLE
Pending cephfs stale cleanup

### ISSUE THREAD

**Issue by Madhu-1** [rook/kubectl-rook-ceph#272]
This tracks the pending work for cephfs stale volumes

* volumes created for NFS csi drivers
* ROX PVC
* stale volumesnapshot cleanup
* Pending clones (?)

@yati1998 above cases are already covered or yet to be covered? Do we have a tracker for it? (i tried to find one but not able to)

**Comment by yati1998** [comment]
1. In the case of pending clones, we won't have subvolumes created at the backend (as per my testing) and hence, I don't think we need to bother about the pending clones. Please correct me if I am wrong.
2. How do we consider a volume snapshot as stale? According to my plan, we are restoring the snapshots so that they can be used to restore the volumes if required.
3. NFS csi drivers are covered with the same logic as a normal csi driver, just an additional step for cleanup of its extra configuration stored needs to be done, we can use this issue itself to track that.
4. Not sure what has to be done for ROX PVC separately!!

**Comment by Madhu-1** [comment]
> 1. In the case of pending clones, we won't have subvolumes created at the backend (as per my testing) and hence, I don't think we need to bother about the pending clones. Please correct me if I am wrong.

Yes it will be created when we create PVC from PVC/Snapshot, please check on this one,

> 2. How do we consider a volume snapshot as stale? According to my plan, we are restoring the snapshots so that they can be used to restore the volumes if required.

Anyone can delete the kubernetes yamls and the backend snapshot might not get deleted.

> 3. NFS csi drivers are covered with the same logic as a normal csi driver, just an additional step for cleanup of its extra configuration stored needs to be done, we can use this issue itself to track that.
 👍🏻 
> 4. Not sure what has to be done for ROX PVC separately!!

This also need to be considered where Ref counters/backend snapshot etc need to be checked and cleaned up if required.

**Comment by yati1998** [comment]
Except the ROX pvc, rest all cases are being covered.

**Comment by yati1998** [comment]
https://github.com/rook/kubectl-rook-ceph/pull/307 
https://github.com/rook/kubectl-rook-ceph/pull/311
https://github.com/rook/kubectl-rook-ceph/pull/312
